### PR TITLE
ambassador: add podSecurityPolicies

### PR DESCRIPTION
**Pull request by 7AC**
#### What this PR does / why we need it:
It allows to bind to external pod security policies without the need to replicate the roles

#### Checklist
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/ambassador/templates/rbac.yaml [hunk 1]
```diff
@@ -29,6 +29,13 @@ rules:
   - apiGroups: [ "apiextensions.k8s.io" ]
     resources: [ "customresourcedefinitions" ]
     verbs: ["get", "list", "watch"]
+  {{- if .Values.podSecurityPolicies }}
```
Comment (Flydiverny) [inline stable/ambassador/templates/rbac.yaml]: Perhaps better to put under rbac.podSecurityPolicies
So it doesn't confuse if we decide to ship a psp with values like podSecurityPolicy.enabled :)

**Comment by Flydiverny** [comment]
Would it be even better if the chart shipped with a psp or would you still want to use your own?

File: stable/ambassador/templates/rbac.yaml (hunk 1 above)
Comment (7AC) [reply stable/ambassador/templates/rbac.yaml]: Sounds good

**Comment by 7AC** [comment]
> Would it be even better if the chart shipped with a psp or would you still want to use your own?

Potentially I could reuse it, but there's probably still a case for allowing to override?

**Comment by 7AC** [comment]
Can I get an `/ok-to-test`?

**Comment by Flydiverny** [comment]
Ah thought we merged this
/lgtm 

Should have been a minor instead of patch version but want to squeeze it in before #15538

**Pull request merged**